server/sonar-db-dao/src/main/java/org/sonar/db/loadedtemplate/LoadedTemplateDto.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.loadedtemplate;
+
+import java.util.Objects;
+
+public final class LoadedTemplateDto {
+
+  public static final String DASHBOARD_TYPE = "DASHBOARD";
+  public static final String FILTER_TYPE = "FILTER";
+  public static final String QUALITY_PROFILE_TYPE = "QUALITY_PROFILE";
+  public static final String PERMISSION_TEMPLATE_TYPE = "PERM_TEMPLATE";
+  public static final String QUALITY_GATE_TYPE = "QUALITY_GATE";
+  public static final String ONE_SHOT_TASK_TYPE = "ONE_SHOT_TASK";
+  public static final String ISSUE_FILTER_TYPE = "ISSUE_FILTER";
+
+  private Long id;
+  private String key;
+  private String type;
+
+  public LoadedTemplateDto() {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1186)